webpack.common.js

```diff
@@ -1,23 +1,25 @@
-const path = require('path');
+const path = require("path");
 
 module.exports = {
-    entry: './src/index.ts',
+    entry: "./src/index.ts",
     module: {
         rules: [
             {
                 test: /\.tsx?$/,
-                use: 'ts-loader',
-                exclude: /node_modules/,
-            },
-        ],
+                use: "ts-loader",
+                exclude: /node_modules/
+            }
+        ]
     },
     resolve: {
-        extensions: ['.ts', '.js']
+        extensions: [".ts", ".js"]
     },
     output: {
-        filename: 'main.js',
-        path: path.resolve(__dirname, 'dist'),
-        library: 'openInsights',
-        libraryTarget: 'window',
-    },
+        filename: "main.js",
+        path: path.resolve(__dirname, "dist"),
+        // TODO: May need to change this depending on the permanent name chosen
+        // for the project
+        library: "openInsights",
+        libraryTarget: "window",
```

Comment:
As the core library is meant to be consumed by other libraries (i.e the site owners application) we actually want WebPack to output a module and not a windowed targeted lib. We also need to ensure `src/index.ts` also re-exports modules useful for provider authors, such as `ProviderBase`, `Test`, and `Fetch`.

```suggestion
        module: true,
```

Reply:
Hmmm. I'm starting to think we might not even need WebPack for the core library and getting `tsc` to emit an es6 module to the `dist` folder should be sufficient. Will have a play and report back.